pandas/core/arrays/period.py

```diff
@@ -908,6 +922,19 @@ def period_array(data, freq=None, ordinal=None, copy=False):
     <PeriodArray>
     ['2017', '2018', 'NaT']
     Length: 3, dtype: period[A-DEC]
+
+    Integers that look like years are handled
+
+    >>> period_array([2000, 2001, 2002], freq='D')
+    ['2000-01-01', '2001-01-01', '2002-01-01']
+    Length: 3, dtype: period[D]
```

Comment:
How can this case be distinguished from ordinals?

Reply:
magic? No idea though. That's what scares me about using them. Should I just remove that?

Reply:
Looks like it all comes down to whatever `Period` does on a scalar.

```python
In [7]: pd.Period(999, freq='D')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-cd7bbf684368> in <module>
----> 1 pd.Period(999, freq='D')

~/sandbox/pandas/pandas/_libs/tslibs/period.pyx in pandas._libs.tslibs.period.Period.__new__()
   2448                 value = str(value)
   2449             value = value.upper()
-> 2450             dt, _, reso = parse_time_string(value, freq)
   2451             if dt is NaT:
   2452                 ordinal = iNaT

~/sandbox/pandas/pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_time_string()
    126         yearfirst = get_option("display.date_yearfirst")
    127
--> 128     res = parse_datetime_string_with_reso(arg, freq=freq,
    129                                           dayfirst=dayfirst,
    130                                           yearfirst=yearfirst)

~/sandbox/pandas/pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_datetime_string_with_reso()
    152
    153     if not _does_string_look_like_datetime(date_string):
--> 154         raise ValueError('Given date string not likely a datetime.')
    155
    156     try:

ValueError: Given date string not likely a datetime.

In [8]: pd.Period(1000, freq='D')
Out[8]: Period('1000-01-01', 'D')

```